backend/src/historicData/service/historicDataQueryRunner.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { isDate } from "date-fns";
+import { Repository } from "typeorm";
+import { UnavailableReason, UnavailableStats } from "../dto/maybeStats.dto";
+import { DelayEntry } from "../entity/delayEntry.entity";
+
+export interface QueryOptions {
+  since?: Date;
+}
+
+const QUERY_OPTIONS_SINCE_DEFAULT = "epoch";
+
+export interface QueryResult {}
+
+/*
+ * Error-state results.
+ */
+
+export const NO_DATA_RESULT: UnavailableStats = {
+  status: "unavailable",
+  reason: UnavailableReason.noData
+};
+
+const INTERNAL_ERROR_RESULT: UnavailableStats = {
+  status: "unavailable",
+  reason: UnavailableReason.internalError
+};
+
+/*
+ * Definition of SQL queries.
+ */
+
+export const DELAY_AT_STATION_QUERY = `
+    WITH historic_with_delay AS (SELECT *, EXTRACT(EPOCH FROM (estimated - planned)::INTERVAL) / 60 AS delay
+                                 FROM historic_data
+                                 WHERE estimated IS NOT NULL
+                                   AND trip_id = $1
+                                   AND stop_id IN (SELECT DISTINCT stop_id
+                                                   FROM stops
+                                                   WHERE "NVBW_HST_DHID" = $2)
+                                   AND planned > $3),
+
+         latest_of_day_tripcode AS (SELECT DISTINCT ON (planned:: date, trip_code) *
+                                    FROM historic_with_delay
+                                    ORDER BY planned:: date, trip_code DESC)
+
+    SELECT MAX(delay), MIN(delay), AVG(delay), STDDEV(delay)
+    FROM latest_of_day_tripcode;
+`;
+
+export const INTERCHANGE_REACHABLE_QUERY = `
+
+`; // TODO
```

Comment:
I suspect these TODOS here are okay for now?

Reply:
Yes, thought it is a good time to merge alle the changes so far (as base for additional stats) before working on the sql queries itself. (At the moment UnavaibleStats is getting returned for interchange reachable and cancellation stats. These TODOs should be gone in second part/PR of this issue.